main.py

```diff
@@ -51,8 +52,11 @@ def start_server(self, socket=os_socket):
                             self.message_queue.put(data)
 
                             if not self.message_queue.empty():
-                                message = self.message_queue.get()
-                                conn.sendall(message)
+                                msg = self.message_queue.get()
+
+                                ack = bytes(json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}), encoding='utf-8')
```

Comment:
I think your code works fine, but I think more common way to convert to bytes is to use the encode() function like this.
```suggestion
                                ack = json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}).encode()
```